### PR TITLE
Mario/ffi without tests

### PR DESCRIPTION
**Pull request by blamario**

### CODE REVIEW

File: Plutarch/FFI.hs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  PlutusTxInner ForallPhantom _ = ForallPhantom
+  PlutusTxInner (a -> b) x = PlutusTxInner a b -> PlutusTxInner b x   -- hack to support Scott encodings
+  PlutusTxInner a x = PlutusTxInner (ScottFn (ScottList (SOP.Code a) x) x) x
+
+{- |
+  List of scott-encoded constructors of a Haskell type (represented by 'SOP.Code')
+
+  ScottList (Code (Either a b)) c = '[a -> c, b -> c]
+-}
+type ScottList :: [[Type]] -> Type -> [Type]
```
Comment (L-as) [inline Plutarch/FFI.hs:76]: Don't we already have this elsewhere?

Comment (srid) [reply Plutarch/FFI.hs:76]: https://github.com/Plutonomicon/plutarch/blob/master/Plutarch/Internal/PlutusType.hs#L298

That one's for Plutarch types, however. This one seems to be for Haskell types.

Comment (blamario) [reply Plutarch/FFI.hs:76]: Yup. The existing one is not reusable here.

File: Plutarch/FFI.hs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  PlutarchInner (PDelayed a) x = PlutarchInner a x
+  PlutarchInner p x = PlutarchInner (PInner p x) x
+
+type family PlutusTxInner (t :: Type) (any :: Type) :: Type where
+  PlutusTxInner Bool _ = Bool
+  PlutusTxInner Integer _ = Integer
+  PlutusTxInner BuiltinString _ = Text
+  PlutusTxInner ForallPhantom _ = ForallPhantom
+  PlutusTxInner (a -> b) x = PlutusTxInner a b -> PlutusTxInner b x   -- hack to support Scott encodings
+  PlutusTxInner a x = PlutusTxInner (ScottFn (ScottList (SOP.Code a) x) x) x
```
Comment (srid) [inline Plutarch/FFI.hs:69]: Doesn't `ScottList (ToPType2 (SOP.Code a))` allow us to reuse the existing type families from `PlutusType.hs`?

**Comment by L-as** [comment]
I discussed this with Mario off GH, and I personally think that we should give up on translating arbitrary types between PlutusTx and Plutarch code. I think the most realistic thing to do is to cover the built-in types only. That would unfortunately exclude booleans, but booleans are Scott-encoded by PlutusTx. We shouldn't rely on the encoding either. The encoding might very well change.

**Pull request closed without merging**